internal/k3s/common.go

```diff
@@ -0,0 +1,167 @@
+package k3s
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	"golang.org/x/mod/semver"
+)
+
+func serviceCmd(action string) *exec.Cmd {
+	var cmd *exec.Cmd
+	if hasSystemd() {
+		cmd = exec.Command("systemctl", action, "k3s")
+	} else {
+		cmd = exec.Command("service", "k3s", action)
+	}
+	return cmd
+}
+
+func hasK3S() bool {
+	_, err := os.Stat(k3sBinary)
+	if err == nil {
+		return true
+	}
+	if err != nil && !os.IsNotExist(err) {
+		fmt.Println("os.Stat: ", err)
+		os.Exit(255)
```

Comment:
minor: I don't like the fact that we invoke os.Exit in a subfunction, can you explain?

Reply:
that's something really rare happened on the server and we can't ignore that. But better then panic :) 
We can move it to just error I guess, you know, it's some difference to have 
```
if hasK3S() {}
``` 
versus 
```
has, err := hasK3S()
if err != nil {}
if has {}
```

Reply:
Agree with that, let's leave it a is. Maybe a comment would be helpful here